lib/model/model.go

```diff
@@ -1346,6 +1346,9 @@ func (m *model) ensureIndexHandler(conn protocol.Connection) *indexHandlerRegist
 	deviceID := conn.DeviceID()
 	connID := conn.ConnectionID()
 
+	// We must acquire fmut first when acquiring both locks.
```

Comment:
Yes but does `RLock` count as acquiring? Gut feeling says no...?

Reply:
I don't see why not? A read-lock can still produce a deadlock when race with a fmut-pmut (full) lock in another function, if the first (full lock) is acquired in both functions and then the subsequent locks (whether read or full) are blocking.

Reply:
Yeah, I wasn't thinking "does the order matter" but rather "is an RLock enough to prevent deadlocks", but I see we already to precisely this thing in at least one other place and I can't think of an ordering that would cause a deadlock.

Reply:
Oh now I get - I indeed haven't thought about that just assumed it's ok. And I too can't think of how only acquiring a read-lock would cause a deadlock (not that this is a very strong indicator that there is none ':) ).